models/cluster.go

```diff
@@ -82,15 +128,27 @@ func (c *Cluster) Create(clientset kubernetes.Interface) error {
 	for _, deployment := range c.Deployments {
```

Comment:
Can you confirm the order that you are creating resources?
For me, It is better to always create service deployments and services and only after success, create app deployments and services.
The reason is that apps will use services, then, let's say app1 uses postgres, if postgres is not ready by the time app1 runs, it will stay in crashloopbackoff
Also, we may need time to setup services like creating and migrating app1 database before it runs...